### PR TITLE
Reload trains last routes when entering run routes

### LINKED ISSUE #286: Display/save previously ran routes
Somewhat related to #283.

What if the routes were depicted by large dot at the end of route and solid line between those dots. Multiple trains would be somewhat simple to differentiate with the large dots. This dot - line - dot could be the color of the company and visible / togleable during opponent turns.

Then if this route is cut with a token, the rest of the route could turn red, meaning it has to be planned again.

Other could see what you ran instead of trying to parse the log.

### PR DESCRIPTION
**Pull request by jenf**
fixes #286 

There's no button to reload the route, you can refresh the browser to do that if someone so desires

### CODE REVIEW

File: lib/engine/corporation.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@ def initialize(sym:, name:, tokens:, **opts)
       @trains = []
       @companies = []
       @revenue_history = {}
+      @last_run = nil
```
Comment (tobymao) [inline lib/engine/corporation.rb:36]: can we store all the runs and replace revenue history with this

File: lib/engine/corporation.rb [hunk 2]
```diff
@@ -102,6 +103,12 @@ def corporation?
       true
     end
 
+    def add_run!(run)
```
Comment (tobymao) [inline lib/engine/corporation.rb:106]: probably need to do like OR, Train, Hexes

File: assets/app/view/route_selector.rb [hunk 3]
```diff
@@ -17,9 +17,32 @@ def active_routes
...
+        end
+        # A future enhancement to this could be to find trains and move the routes over
+        Engine::Route.new(@game.phase, train, connection_hexes: connection_hexes) if trains.include?(train)
+      end.compact
+    end
+
     def render
       trains = @game.round.current_entity.trains
 
+      if @routes.empty?
```
Comment (tobymao) [inline assets/app/view/route_selector.rb:37]: this looks awkward, the code will probably change when we move last_revenue to all revenues but if you were to keep this

it should be like

@routes = last_run if @routes.empty

...

File: lib/engine/corporation.rb (hunk 1 above)
Comment (jenf) [reply lib/engine/corporation.rb:36]: yeah, but we need to store the connection_hexes in textual form due to upgrades on tiles.

I'll add a new class to store this info and replace it with @operating_history

File: assets/app/view/route_selector.rb (hunk 3 above)
Comment (jenf) [reply assets/app/view/route_selector.rb:38]: You'd need to store the routes and selected routes so the map updates correctly.

File: lib/engine/corporation.rb (hunk 1 above)
Comment (tobymao) [reply lib/engine/corporation.rb]: you can store both the text form and maybe a property for revenue

Comment (jenf) [reply lib/engine/corporation.rb]: yep, see the diff :)

**Pull request merged**